parsers/OPENNEM.py

```diff
@@ -0,0 +1,216 @@
+            'oil': sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES['oil']),
+            'hydro': sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES['hydro']),
+            'wind': sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES['wind']),
+            'biomass': sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES['biomass']),
+            # We here assume all rooftop solar is fed to the grid
+            # This assumption should be checked and we should here only report
+            # grid-level generation
+            'solar': sum_vector(row, OPENNEM_PRODUCTION_CATEGORIES['solar']),
+        },
+        'storage': {
+            # opennem reports charging as negative, we here should report as positive
+            # Note: we made the sign switch before, so we can just sum them up
+            'battery': sum_vector(row, OPENNEM_STORAGE_CATEGORIES['battery']),
+            # opennem reports pumping as positive, we here should report as positive
+            'hydro': sum_vector(row, OPENNEM_STORAGE_CATEGORIES['hydro']),
+        },
+        'capacity': {
+            'coal': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['coal']),
+            'gas': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['gas']),
+            'oil': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['oil']),
+            'hydro': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['hydro']),
+            'wind': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['wind']),
+            'biomass': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['biomass']),
+            'solar': sum_vector(capacities, OPENNEM_PRODUCTION_CATEGORIES['solar']),
+        },
+        'source': SOURCE,
+        'zoneKey': zone_key,
+    } for dt, row in df.iterrows()]
+
+    # Validation
+    for obj in objs:
+        for k, v in obj['production'].items():
+            if v is None:
+                continue
+            if v < 0 and v > -50:
+                # Set small negative values to 0
+                logger.warning(f'Setting small value of {k} ({v}) to 0.',
+                               extra={'key': zone_key})
+                obj['production'][k] = 0
+
+    return objs
+
+
+def fetch_price(zone_key=None, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    df = fetch_main_df('price', zone_key=zone_key, session=session, target_datetime=target_datetime, logger=logger)
+    df = df.loc[~df['PRICE'].isna()]  # Only keep prices that are defined
+    return [{
+        'datetime': arrow.get(dt.to_pydatetime()).datetime,
+        'price': sum_vector(row, ['PRICE']),  # currency / MWh
+        'currency': 'AUD',
+        'source': SOURCE,
+        'zoneKey': zone_key,
+    } for dt, row in df.iterrows()]
+
+
+def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    sorted_zone_keys = sorted([zone_key1, zone_key2])
+    key = '->'.join(sorted_zone_keys)
+    # Remember to reverse flow
```

Comment:
The comment confuses me. Did we remember this?

Reply:
indeed it was done. Good catch.